From 3eda836d2e37f79076e38c48936c64485bea678d Mon Sep 17 00:00:00 2001
From: Prabhat Chand Pandey <prabhat.chand.pandey@intel.com>
Date: Wed, 3 Apr 2019 18:38:38 +0530
Subject: [PATCH] Device : Intel : Project-Celadon : Adding Sepolicy for DBC

Adding sepolicy for adb over DBC.
This adds the sepolicy for Celadon KBL and APL NUC.
It also have support for dual XHCI also.

Tracked-On:
Signed-off-by: Prabhat Chand Pandey <prabhat.chand.pandey@intel.com>
---
 sepolicy/file.te                  | 1 +
 sepolicy/usb-gadget/adbd.te       | 1 +
 sepolicy/usb-gadget/device.te     | 1 +
 sepolicy/usb-gadget/file_contexts | 7 +++++++
 sepolicy/usb-gadget/init.te       | 2 ++
 5 files changed, 12 insertions(+)
 create mode 100644 sepolicy/usb-gadget/adbd.te
 create mode 100644 sepolicy/usb-gadget/device.te
 create mode 100644 sepolicy/usb-gadget/file_contexts
 create mode 100644 sepolicy/usb-gadget/init.te

diff --git a/sepolicy/file.te b/sepolicy/file.te
index 2d0967a..e4a68e8 100644
--- a/sepolicy/file.te
+++ b/sepolicy/file.te
@@ -5,4 +5,5 @@ type sysfs_thermal_writable, fs_type, sysfs_type;
 type atvr_device, dev_type;
 type sysfs_coretemp, fs_type, sysfs_type;
 type gpu_pid_stats_file, fs_type, debugfs_type;
+type dbc_sysfs, sysfs_type, fs_type;
 type debugfs_pstate, fs_type, debugfs_type;
diff --git a/sepolicy/usb-gadget/adbd.te b/sepolicy/usb-gadget/adbd.te
new file mode 100644
index 0000000..93133a0
--- /dev/null
+++ b/sepolicy/usb-gadget/adbd.te
@@ -0,0 +1 @@
+allow adbd dbc_device:chr_file rw_file_perms;
diff --git a/sepolicy/usb-gadget/device.te b/sepolicy/usb-gadget/device.te
new file mode 100644
index 0000000..9c131c9
--- /dev/null
+++ b/sepolicy/usb-gadget/device.te
@@ -0,0 +1 @@
+type dbc_device, dev_type;
diff --git a/sepolicy/usb-gadget/file_contexts b/sepolicy/usb-gadget/file_contexts
new file mode 100644
index 0000000..b22ab88
--- /dev/null
+++ b/sepolicy/usb-gadget/file_contexts
@@ -0,0 +1,7 @@
+/dev/dbc_raw	u:object_r:dbc_device:s0
+/sys/devices/pci0000:00/0000:00:14.0/dbc	u:object_r:dbc_sysfs:s0
+/sys/bus/pci/devices/0000:00:14.0/dbc		u:object_r:dbc_sysfs:s0
+/sys/devices/pci0000:00/0000:00:15.0/dbc	u:object_r:dbc_sysfs:s0
+/sys/bus/pci/devices/0000:00:15.0/dbc		u:object_r:dbc_sysfs:s0
+/sys/devices/pci0000:00/0000:00:39.0/dbc	u:object_r:dbc_sysfs:s0
+/sys/bus/pci/devices/0000:00:39.0/dbc		u:object_r:dbc_sysfs:s0
diff --git a/sepolicy/usb-gadget/init.te b/sepolicy/usb-gadget/init.te
new file mode 100644
index 0000000..a306968
--- /dev/null
+++ b/sepolicy/usb-gadget/init.te
@@ -0,0 +1,2 @@
+allow init dbc_device:chr_file rw_file_perms;
+allow init dbc_sysfs:file rw_file_perms;
--
2.21.0
